tests/reset/reset.cpp

```diff
@@ -201,203 +166,69 @@ TEST(Reset, printResetWithVariable)
-        "</model>\n";
-    libcellml::ModelPtr m = createModelWithComponent();
-    libcellml::ComponentPtr c = m->component(0);
-    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
-
-    libcellml::WhenPtr w1 = std::make_shared<libcellml::When>();
-    libcellml::WhenPtr w2 = std::make_shared<libcellml::When>();
-    libcellml::WhenPtr w3 = std::make_shared<libcellml::When>();
-
-    r->addWhen(w1);
-    r->addWhen(w2);
-    r->addWhen(w3);
-    c->addReset(r);
-
-    libcellml::Printer p;
-
-    const std::string a = p.printModel(m);
-    EXPECT_EQ(e, a);
-}
+    std::string a;
+    static const std::string emptyMath =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "        </math>\n";
 
-TEST(Reset, printResetWithWhenWithValueSet)
-{
     const std::string e =
         "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
-        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
-        "  <component>\n"
-        "    <reset>\n"
-        "      <when>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"model\">\n"
+        "  <component name=\"component\">\n"
+        "    <variable name=\"variable1\"/>\n"
+        "    <variable name=\"variable2\"/>\n"
+        "    <reset variable=\"variable1\" test_variable=\"variable2\" order=\"1\">\n"
+        "      <reset_value>\n"
         "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
-        "          a value set\n"
         "        </math>\n"
-        "      </when>\n"
+        "      </reset_value>\n"
         "    </reset>\n"
         "  </component>\n"
         "</model>\n";
-    const std::string math =
-        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
-        "  a value set\n"
-        "</math>\n";
 
     libcellml::ModelPtr m = createModelWithComponent();
     libcellml::ComponentPtr c = m->component(0);
-    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
-    libcellml::WhenPtr w = std::make_shared<libcellml::When>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    libcellml::ResetPtr r1 = std::make_shared<libcellml::Reset>();
+    libcellml::ResetPtr r2 = std::make_shared<libcellml::Reset>();
+    libcellml::ResetPtr r3 = std::make_shared<libcellml::Reset>();
+    libcellml::Printer printer;
 
-    w->setValue(math);
-    r->addWhen(w);
-    c->addReset(r);
+    m->setName("model");
+    c->setName("component");
+    v1->setName("variable1");
+    v2->setName("variable2");
 
-    libcellml::Printer p;
+    r1->setVariable(v1);
+    r1->setTestVariable(v2);
+    r1->setOrder(1);
+    r1->setResetValue(emptyMath);
 
-    const std::string a = p.printModel(m);
+    c->addReset(r1);
+    c->addVariable(v1);
+    c->addVariable(v2);
+
+    a = printer.printModel(m);
     EXPECT_EQ(e, a);
 }
 
-TEST(Reset, printResetWithMultipleWhensWithValues)
+TEST(Reset, printResetWithOrder)
```

Comment:
would be good to add tests with negative orders.

Reply:
There will be more tests around orders to come when the order checking is in properly - at the moment in this PR its scope is not correct. Related to issue #326 (and discussion earlier in this PR).

Reply:
Order validation is now present in the other PR #350.  Tests will need to be tweaked when this is merged in, so if it's ok will add more tests when they don't need rewriting?

Reply:
I've added a test with a negative order to the validator in this branch.